node/src/components/block_validator.rs

```diff
@@ -59,114 +68,114 @@ pub(crate) struct BlockValidationState<T> {
     responders: SmallVec<[Responder<(bool, T)>; 2]>,
 }
 
-/// Block validator.
-#[derive(DataSize, Debug, Default)]
-pub(crate) struct BlockValidator<T, I> {
+#[derive(Debug)]
+pub(crate) struct BlockValidatorReady<T, I> {
+    /// Chainspec loaded for deploy validation.
+    chainspec: Arc<Chainspec>,
```

Comment:
`BlockValidator` shouldn't need the whole chainspec, it has things like Genesis contracts (those are byte arrays). It should have a smaller, subset of configuration parameters that it needs. Can be ticketed and done separately.

Reply:
Another thing, this configuration will change based on the protocol version so it cannot be set in stone. Block Validator may have to use different configurations for block #11, and later block #12 and may even have to go back to the configuration from the block #11 if the messages arrive out of order (from other nodes, not internal ones).

Reply:
If I'm not mistaken, this is just an Arc reference to a shared value, so we aren't copying those byte arrays around. It seems like the same is done in the `block_proposer` as well.

Reply:
Might be, we should change it in both places then.

Reply:
> Another thing, this configuration will change based on the protocol version so it cannot be set in stone. Block Validator may have to use different configurations for `block #11`, and later `block #12` and may even have to go back to the configuration from the `block #11` if the messages arrive out of order (from other nodes, not internal ones).

Yes, this should be handled when we work on chainspec upgrades.

Reply:
Is there a ticket for that so that we don't lose the track of it?

Reply:
I mentioned this to @Fraser999, he's working on designing our upgrade approach.

Reply:
Consensus has [`ProtocolConfig`](https://github.com/CasperLabs/casper-node/blob/master/node/src/components/consensus/config.rs#L27) that uses a subset of the original `Chainspec` and is actually [created from one](https://github.com/CasperLabs/casper-node/blob/master/node/src/components/consensus/config.rs#L41).

Reply:
Side-note: The `Arc` is mostly there for transfering the chainspec around. I just did not feel like cloning it each time it was requested. I used a `ProtocolConfig` because that is what the code previously used, so I am currently neither for nor against keeping the actual `Arc` around :)